src/ode_solver/JFNK_solver/jacobian_vector_product.h

```diff
@@ -0,0 +1,59 @@
+#ifndef __JACOBIAN_VECTOR_PRODUCT__
+#define __JACOBIAN_VECTOR_PRODUCT__
+
+#include "dg/dg.h"
+
+namespace PHiLiP {
+namespace ODE{
+
+//template things
+//UPDATE WITH #IF STUFF
+template <int dim, typename real, typename MeshType>
+class JacobianVectorProduct{
+public:
+
+    JacobianVectorProduct(std::shared_ptr< DGBase<dim, real, MeshType> > dg_input);
+
+    ~JacobianVectorProduct() {};
+
+    void reinit_for_next_timestep(double dt_input,
+                double epsilon_input,
+                dealii::LinearAlgebra::distributed::Vector<double> &previous_step_solution_input);
+
+    void reinit_for_next_Newton_iter(dealii::LinearAlgebra::distributed::Vector<double> &current_solution_estimate_input);
+
+    // Application of matrix to vector src. Write result into dst.
+    void vmult (dealii::LinearAlgebra::distributed::Vector<double> &dst,
+                const dealii::LinearAlgebra::distributed::Vector<double> &src) const;
```

Comment:
would it be possible to give more descriptive variable names than `dst` and `src` ?

Reply:
I can expand `dst` to `destination` (and likewise for the other function using that naming). I will change `src` to `w` and modify the comment to make it more clear that the Jacobian-vector product is returned. I originally named to be consistent with `deal.ii` conventions, but in this context it's a little more clear to change to more verbose naming.